base/doc/usrguide.tex

```diff
@@ -1258,4 +1258,39 @@ \section{Case changing}
   |\DeclareUppercaseMapping[xx]{"0131}{I}|
 \end{quote}
 
+\section{Support for problem solving}
+
+\begin{decl}
+  |\listfiles| \oarg{options}
+\end{decl}
+
+If this command is placed in the preamble then a list of the files
+read in (as a result of processing the document) will be displayed
+on the terminal (and in the log file) at the end of the run. Where
+possible, a short description will also be produced. These descriptions
+will (hopefully) include the descriptions, dates and version numbers
+for package and class files.
+
+Sometimes, if may be that a local edit has been made to a package or
+class file (or rather a copy of such a file). To allow these cases to
+be identified, \cs{listfiles} takes an optional argument which allows
+adjustment of the information printed using a key--value approach
+\begin{description}
+  \item[\texttt{hashes}] Adds the MD5 hash for each file to the
+    information printed
+  \item[\texttt{sizes}] Adds the file size for each file to the
+    information printed
```

Comment:
There was the suggestion to also offer "paths". I have my doubts that it is that helpful, because if somebody sends me his output that includes full paths of filename then I can't do a simple diff to what I get. So for that use case it would be counterproductive I think. But I can see that locally it might be helpful, so I'm not sure if that should be included.

I suggested "titles=false" as a further possibility, but thinking about it I feel this only make things more complex without any real gain, so I guess I take that back.

Reply:
We can revisit this if there is a clear benefit, I guess: that's why I went keyval rather than a simple `\listfilehashes`